cmd/2016_1/main.go

```diff
@@ -0,0 +1,113 @@
+package main
+
+import (
+	"container/ring"
+	"flag"
+	"image"
+	"log"
+	"strconv"
+	"strings"
+
+	"github.com/afarbos/aoc/pkg/point"
+	"github.com/afarbos/aoc/pkg/read"
+	"github.com/afarbos/aoc/pkg/utils"
+)
+
+const separator = ", "
+
+var flagInput string
+
+func init() {
+	utils.Init(&flagInput)
+}
+
+type person struct {
+	firstCross, position image.Point
+	direction            *ring.Ring
+	visited              map[image.Point]struct{}
+}
+
+func newPerson(sequences []string) *person {
+	var (
+		p                                        = new(person)
+		northRing, eastRing, southRing, westRing = ring.New(1), ring.New(1), ring.New(1), ring.New(1)
+	)
+
+	northRing.Value = image.Point{1, 0}
+	eastRing.Value = image.Point{0, 1}
+	southRing.Value = image.Point{-1, 0}
+	westRing.Value = image.Point{0, -1}
+
+	southRing.Link(westRing)
+	eastRing.Link(southRing)
+	northRing.Link(eastRing)
+
+	p.direction = northRing
+	p.visited = make(map[image.Point]struct{}, len(sequences)*2)
+
+	p.walkBlocks(sequences)
+
+	return p
+}
+
+func (p *person) walk(newDirection byte, blockCount int) {
+	const (
+		right = "R"
+		left  = "L"
+	)
+
+	switch newDirection {
+	case right[0]:
+		p.direction = p.direction.Next()
+	case left[0]:
+		p.direction = p.direction.Prev()
+	default:
+		log.Fatal(newDirection)
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [calls to log.Fatal only in main() or init() functions](https://app.codacy.com/manual/afarbos/aoc/pullRequest?prid=5498955)